docs/root/operations/admin.rst

```diff
@@ -439,11 +439,18 @@ modify different aspects of the server:
 
   .. http:get:: /stats?filter=regex
 
-  Filters the returned stats to those with names matching the regular expression
-  ``regex``. Compatible with ``usedonly``. Performs partial matching by default, so
-  ``/stats?filter=server`` will return all stats containing the word ``server``.
-  Full-string matching can be specified with begin- and end-line anchors. (i.e.
-  ``/stats?filter=^server.concurrency$``)
+  Filters the returned stats to those with names matching the regular
+  expression ``regex`` using Ecmascript syntax. Compatible with
+  ``usedonly``. Performs partial matching by default, so
+  ``/stats?filter=server`` will return all stats containing the word
+  ``server``.  Full-string matching can be specified with begin- and
+  end-line anchors. (i.e.  ``/stats?filter=^server.concurrency$``)
+
+  .. http:get:: /stats?filter=regex&safe
```

Comment:
Would it make more sense to be `&engine=safe` (or maybe `&engine=re2`), so that `&engine=std` can also be specified?

Reply:
I would probably go with engine=re2 and engine=std, but stepping back, do we actually have to support std at all? Do you think anyone actually cares about this? Can we just switch it and put it behind a runtime flag? I really doubt anyone will complain.

Reply:
How different are the std::regex-ecmascript and re2 languages? I'm not a regex wizard by any stretch, but looking at the re2 syntax docs and emcascript docs, all the normal stuff I've ever tried in a regex looks the same.

So I'd be in favor of just swapping it out with runtime override.

Reply:
Let's try runtime first. If someone complains we can add config. I will be shocked if this causes an issue with the basic regex that people are likely to use.

Reply:
Easy enough. Do you think it should default to RE2 or should we initially in this PR have it default off?

Reply:
Let's default to re2 and see if anyone complains.

Reply:
done